jablib/src/test/java/org/jabref/logic/git/GitHandlerTest.java

```diff
@@ -1,29 +1,61 @@
 package org.jabref.logic.git;
 
 import java.io.IOException;
+import java.net.URISyntaxException;
 import java.nio.file.Files;
 import java.nio.file.Path;
 import java.util.Iterator;
+import java.util.Optional;
 
 import org.eclipse.jgit.api.Git;
 import org.eclipse.jgit.api.errors.GitAPIException;
 import org.eclipse.jgit.lib.AnyObjectId;
 import org.eclipse.jgit.lib.Constants;
 import org.eclipse.jgit.revwalk.RevCommit;
+import org.eclipse.jgit.transport.RefSpec;
+import org.eclipse.jgit.transport.URIish;
 import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
 import org.junit.jupiter.api.io.TempDir;
 
 import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
 
 class GitHandlerTest {
     @TempDir
     Path repositoryPath;
+    @TempDir
+    Path remoteRepoPath;
+    @TempDir
+    Path clonePath;
     private GitHandler gitHandler;
 
     @BeforeEach
-    void setUpGitHandler() {
+    void setUpGitHandler() throws IOException, GitAPIException, URISyntaxException {
         gitHandler = new GitHandler(repositoryPath);
+
+        Git remoteGit = Git.init()
+                           .setBare(true)
+                           .setDirectory(remoteRepoPath.toFile())
+                           .call();
+        Path testFile = repositoryPath.resolve("initial.txt");
+        Files.writeString(testFile, "init");
+
+        gitHandler.createCommitOnCurrentBranch("Initial commit", false);
+
+        try (Git localGit = Git.open(repositoryPath.toFile())) {
+            localGit.remoteAdd()
+                    .setName("origin")
+                    .setUri(new URIish(remoteRepoPath.toUri().toString()))
+                    .call();
+
+            localGit.push()
+                    .setRemote("origin")
+                    .setRefSpecs(new RefSpec("refs/heads/main:refs/heads/main"))
+                    .call();
+        }
+
+        Files.writeString(remoteRepoPath.resolve("HEAD"), "ref: refs/heads/main");
```

Comment:
Reads a bit hacky... https://stackoverflow.com/a/33102959/873282 reads verbose, thus it is OK for me.